storage/localstore/mode_set_test.go

```diff
@@ -63,180 +64,341 @@ func TestModeSetAccess(t *testing.T) {
-			expErrPushIndex: leveldb.ErrNotFound, // not in push index
-			expErrGCIndex:   nil,                 // GCd
-			expErrPinIndex:  leveldb.ErrNotFound, // not pinned
-		},
-		// tag IS NOT anonymous, set push synced WILL cause it to go
-		// into gc index, it SHOULD be in the pin index
-		{
-			name:            "set push sync, normal tag, with pinning",
-			anonymous:       false,
-			pin:             true,
-			mode:            chunk.ModeSetSyncPush,
-			expErrPushIndex: leveldb.ErrNotFound, // not in push index
-			expErrGCIndex:   nil,                 // GCd
-			expErrPinIndex:  nil,                 // is pinned
-		},
-		// tag IS anonymous, set push synced WILL cause it to go
-		// into gc index, it SHOULD NOT be in the pin index
-		{
-			name:            "set push sync, anonymous tag, no pinning",
-			anonymous:       true,
-			pin:             false,
-			mode:            chunk.ModeSetSyncPush,
-			expErrPushIndex: nil,                 // not in push index
-			expErrGCIndex:   leveldb.ErrNotFound, // not GCd
-			expErrPinIndex:  leveldb.ErrNotFound, // not pinned
-		},
-		// tag IS anonymous, set push synced WILL NOT cause it to go
-		// into gc index, it SHOULD be in the pin index
-		{
-			name:            "set push sync, anonymous tag, with pinning",
-			anonymous:       true,
-			pin:             true,
-			mode:            chunk.ModeSetSyncPush,
-			expErrPushIndex: nil,                 // is in push index
-			expErrGCIndex:   leveldb.ErrNotFound, // not GCd
-			expErrPinIndex:  nil,                 // is pinned
-		},
-	} {
-		t.Run(mtc.name, func(t *testing.T) {
-			for _, tc := range multiChunkTestCases {
-				t.Run(tc.name, func(t *testing.T) {
-					db, cleanupFunc := newTestDB(t, &Options{Tags: chunk.NewTags()})
-					defer cleanupFunc()
-
-					tag, err := db.tags.Create(mtc.name, int64(tc.count), mtc.anonymous)
-					if err != nil {
-						t.Fatal(err)
-					}
-					if tag.Uid != 0 {
-						t.Fatal("expected mock tag uid")
-					}
+// here we try to set a normal tag (that should be handled by pushsync)
+// as a result we should expect the tag value to remain in the pull index
+// and we expect that the tag should not be incremented by pull sync set
+func TestModeSetSyncPullNormalTag(t *testing.T) {
+	db, cleanupFunc := newTestDB(t, &Options{Tags: chunk.NewTags()})
+	defer cleanupFunc()
 
-					chunks := generateTestRandomChunks(tc.count)
+	tag, err := db.tags.Create("test", 1, false)
+	if err != nil {
+		t.Fatal(err)
+	}
 
-					wantTimestamp := time.Now().UTC().UnixNano()
-					defer setNow(func() (t int64) {
-						return wantTimestamp
-					})()
+	ch := generateTestRandomChunk().WithTagID(tag.Uid)
+	_, err = db.Put(context.Background(), chunk.ModePutUpload, ch)
+	if err != nil {
+		t.Fatal(err)
+	}
 
-					_, err = db.Put(context.Background(), chunk.ModePutUpload, chunks...)
-					if err != nil {
-						t.Fatal(err)
-					}
+	tag.Inc(chunk.StateStored) // so we don't get an error on tag.Status later on
```

Comment:
If this is needed in the test, should it be the responsibility of db.Put to increment StateStored if it finds the tag?

Reply:
no, StateStored is incremented in `hasherstore`..... remove this line and you will understand the error that is invoked. this must be called in the test because of the `tag.Status` logic. please review that function to understand why this is called

Reply:
Thanks for the explanation.